packages/aws-cdk-lib/aws-rds/lib/cluster.ts

```diff
@@ -667,6 +696,7 @@ abstract class DatabaseClusterNew extends DatabaseClusterBase {
       associatedRoles: clusterAssociatedRoles.length > 0 ? clusterAssociatedRoles : undefined,
       deletionProtection: defaultDeletionProtection(props.deletionProtection, props.removalPolicy),
       enableIamDatabaseAuthentication: props.iamAuthentication,
+      enableHttpEndpoint: Lazy.any({ produce: () => this.enableDataApi }),
```

Comment:
Why is this using `lazy.any..` instead of directly `props.enableDataApi`?

Reply:
@GavinZZ 

If we don't use `Lazy` in this point, [this test](https://github.com/aws/aws-cdk/pull/29338/files#diff-5faaee1dfd608a160551274db73518aeab8d44bd3351879b34941a052a0135f9R4108) will be failed `!!   Missing key 'EnableHttpEndpoint'`.
This is because `this.enableDataApi` is modified in the `grantDataApiAccess()`.

Initially, I was surprised by this outcome as well. I had thought that `Lazy` was merely for using CloudFormation's built-in functions. If there's anything I've misunderstood, please don't hesitate to correct me.

Reply:
Thanks for the explanation. I think it's a valid use case here to use `Lazy`.